### DISCUSSION TITLE
Github Actions runners do not detect organisation runners from Public repositories

### DISCUSSION THREAD

**Discussion by SimonDarksideJ** [community/community#25272] [Actions]
<p>A curious issue today while continuing our attempt to utilise runners and retire our Azure pipelines dependencies.</p>
<p>If I setup ORGANISATION runners:</p>
<ul>
<li>They can be seen by PRIVATE repositories created by that organisation - they work fine</li>
</ul>
<p>However</p>
<ul>
<li>If I have a PUBLIC repository in my OWN organisation, that repository is unable to see the runners in my Organisation.</li>
<li>if I fork a repo from another repository to my OWN organisation, that fork is unable to see the runners in my Organisation.</li>
</ul>
<p>If I create a runner inside the public or forked repository for my organisation, then it works fine but refuses to detect the organisation runners?  This makes it very difficult to manage when there are several forked repositories we want to utilise our self hosted runner estate configured already for the organisation.</p>
<p>Is there a way around this?</p>

**Comment by Katsute** [comment]
<p>Having organizational runners visible on forks presents a security risk, since <strong>anyone</strong> can fork a repository.</p>

**Comment by SimonDarksideJ** [comment]
<p>I wholeheartedly disagree.  but I can see where you might have gotten the wrong conclusion from my description, so I have updated the test to be clearer <a class="mention" href="/katsute">@katsute</a></p>
<p>I OWN the organisation, and my ORGANISATION forked the repository.</p>
<p>So why can my organisation not use its runners to build the fork I own in my repository?</p>
<p>I can fully understand not being able to use the runner in the source organisation for the fork.  But it makes no sense for a fork in an organisation not to be able to use the runners OF that organisation?</p>

**Comment by Katsute** [comment]
<p>Maybe this documentation might help <a href="https://docs.github.com/en/actions/hosting-your-own-runners/managing-access-to-self-hosted-runners-using-groups#changing-the-access-policy-of-a-self-hosted-runner-group" class="inline-onebox" rel="noopener nofollow ugc">Managing access to self-hosted runners using groups - GitHub Docs</a>, I don’t use self-hosted runners so I can’t really say for sure how to make forks use org runners.</p>

**Comment by jsoref** [comment]
<p>Do you have a visible example of this?</p>
<p>Usually when we fork public repositories, their workflows ask for github hosted runners, and thus we wouldn’t trip on this.</p>
<p>In fact, it’s pretty annoying to run across public repositories that require self-hosted runners, since it means that when I fork them (and I fork <em>lots</em> of repositories), their workflows won’t work.</p>

**Comment by SimonDarksideJ** [comment]
<p>Yup, got an example right here:<br>
<a href="https://github.com/realitycollective/com.realitytoolkit.core" rel="noopener nofollow ugc">realitycollective/com.realitytoolkit.core: The all-purpose framework for building the next reality (github.com)</a></p>
<p>We have actually replaced the workflows from this fork as we needed to work slightly differently.</p>
<p>But in any case, If there are workflows in a Fork, I agree they shouldn’t run by default and should NOT use the runners from the source repo.</p>
<p>HOWEVER, if I have runners configured in my Organisation and I have enabled them for those Forked repos, then they should be able to run and use my Organisations runners.  I Should not have to create entirely new runner just for that fork in order to Build it.</p>

**Comment by SimonDarksideJ** [comment]
<p>Had to update the post as the situation is much deeper.  PUBLIC repositories of ANY kind are unable to see the runners of an Organisation.</p>
<p>Organisation runners are ONLY usable by private repositories (which is hilarious since reusable workflows HAVE to be in a PUBLIC repository to be used)</p>

**Comment by ChristopherHX** [comment]
<p>Do you have allowed the default runner group for public repositories? GitHub has it disabled by default.<br>
Please go to <code>https://github.com/organizations/:org/settings/actions/runners</code> ,<br>
<a href="https://user-images.githubusercontent.com/44845461/127209360-cadff387-ebca-45fc-8b37-da3634997c8c.png" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/44845461/181090020-a06f7716-8011-4a91-b3f6-e4b4aa872305.png" alt="image" data-base62-sha1="8aGmli4e7nODkK2aPK4xXNujdaY" width="690" height="245"></a><br>
Press the three dots of the default group and change repository access to include public repositories (this implies a potential security risk, due to PR’s)</p>
<p><em>I haven’t tested public forks, so idk if this works after allowing it</em></p>

**Comment by SimonDarksideJ** [comment]
<p>I’ve 3 self hosted runners registered<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/4/34620b6a26eeb3de21af84829643b624cf5530a8.png" data-download-href="/uploads/short-url/7toT3zkagInYFFmIuz2UDn7l9ws.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1793042/181090022-65266e54-6249-411a-8ecd-ca9a72cbe9ee.png" alt="image" data-base62-sha1="7toT3zkagInYFFmIuz2UDn7l9ws" width="690" height="195" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/4/34620b6a26eeb3de21af84829643b624cf5530a8_2_690x195.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/4/34620b6a26eeb3de21af84829643b624cf5530a8_2_1035x292.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/4/34620b6a26eeb3de21af84829643b624cf5530a8.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/4/34620b6a26eeb3de21af84829643b624cf5530a8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1201×340 23.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>If the repository is Private, these are used.  If the repository is PUBLIC, they are ignored</p>

**Comment by ChristopherHX** [comment]
<p>Registering runners is not enough for an org to be usable for public repositories, here also the official documentation how to make public repos work <a href="https://docs.github.com/en/actions/hosting-your-own-runners/managing-access-to-self-hosted-runners-using-groups#changing-the-access-policy-of-a-self-hosted-runner-group" class="inline-onebox" rel="noopener nofollow ugc">Managing access to self-hosted runners using groups - GitHub Docs</a></p>
<p>They changed the UI, but still the same thing you need to do. I’m not shure if that ui change made it even more confusing to enable public repository access of org runners.</p>
<p>You need to open the runner group view<br>
<code>https://github.com/organizations/&lt;org&gt;/settings/actions/runner-groups</code></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/5/f5c4527cf840a58a72e0fe4b72ea4d9c83c7db26.png" data-download-href="/uploads/short-url/z49B7j3EOlJCjjQ3d9hzwXRNxA2.png?dl=1" title="Screenshot_20220301-091839~2" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/44845461/181090026-b0839a29-3820-4515-bf69-346dd6602c6c.png" alt="Screenshot_20220301-091839~2" data-base62-sha1="z49B7j3EOlJCjjQ3d9hzwXRNxA2" width="517" height="132" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/5/f5c4527cf840a58a72e0fe4b72ea4d9c83c7db26_2_517x132.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/5/f5c4527cf840a58a72e0fe4b72ea4d9c83c7db26_2_775x198.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/f/5/f5c4527cf840a58a72e0fe4b72ea4d9c83c7db26_2_1034x264.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/5/f5c4527cf840a58a72e0fe4b72ea4d9c83c7db26_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot_20220301-091839~2</span><span class="informations">2410×619 59.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div><br>
Now click the runner group name, I only have default with the github free plan
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/6/c68917b1fc9142944503d89e3681fadd0fa88ad8.png" data-download-href="/uploads/short-url/skkg24oCc9tpWhxusp59XZTX0rm.png?dl=1" title="Screenshot_20220301-091923~2" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/44845461/181090028-188712e8-4651-4252-927e-faa938617cb8.png" alt="Screenshot_20220301-091923~2" data-base62-sha1="skkg24oCc9tpWhxusp59XZTX0rm" width="517" height="175" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/6/c68917b1fc9142944503d89e3681fadd0fa88ad8_2_517x175.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/6/c68917b1fc9142944503d89e3681fadd0fa88ad8_2_775x262.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/6/c68917b1fc9142944503d89e3681fadd0fa88ad8_2_1034x350.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/6/c68917b1fc9142944503d89e3681fadd0fa88ad8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot_20220301-091923~2</span><span class="informations">2355×801 69.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div><br>
Enable Allow public repositories

**Comment by SimonDarksideJ** [comment]
<p>If Runner Groups are required then that is going to limit it for a lot of people, as Runner Groups are only available for Enterprise accounts.</p>
<p>The org I work for only has a Teams Pro account as its a startup, most Open Source organisations still only use the free tier.</p>
<p>But at least it is an answer <a class="mention" href="/christopherhx">@ChristopherHX</a> thanks</p>

**Comment by ChristopherHX** [comment]
<aside class="quote no-group" data-username="SimonDarksideJ" data-post="11" data-topic="232382">
<div class="title">
<div class="quote-controls"></div>
 SimonDarksideJ:</div>
<blockquote>
<p>If Runner Groups are required then that is going to limit it for a lot of people, as Runner Groups are only available for Enterprise accounts</p>
</blockquote>
</aside>
<p><a class="mention" href="/simondarksidej">@SimonDarksideJ</a> Do you want to tell me you don’t have that settings section? I only see a message that I’m not allowed to <strong>create</strong> a runner group, but there is a runner group called Default which you can edit to allow public repositories to access your organization runners.</p>
<p>I’m using the free tier and it is available for me.</p>

**Comment by SimonDarksideJ** [comment]
<p>Correct, our Organisation cannot create new runner groups<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/4/145c7ff5a80c61b07b96bca204a198a4b251799f.png" data-download-href="/uploads/short-url/2U7JdQ4s24mgpMee2n3E00uJhEP.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1793042/181090032-c35b367b-91c1-45b4-aa33-ab8114832546.png" alt="image" data-base62-sha1="2U7JdQ4s24mgpMee2n3E00uJhEP" width="690" height="159" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/4/145c7ff5a80c61b07b96bca204a198a4b251799f_2_690x159.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/4/145c7ff5a80c61b07b96bca204a198a4b251799f_2_1035x238.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/4/145c7ff5a80c61b07b96bca204a198a4b251799f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/4/145c7ff5a80c61b07b96bca204a198a4b251799f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1187×275 13.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>It’s the same on an opensource project I manage and the option is not available for projects (Runners are only supported by Organisations</p>

**Comment by davidcorbin-atmosera** [comment, marked as answer]
<p><a class="mention" href="/simondarksidej">@SimonDarksideJ</a>  - but what if you try to modify the EXISTING (aka Default) group?</p>

**Comment by SimonDarksideJ** [comment]
<p>I was about to retort, but you are completely right.</p>
<p>Wasn’t even aware the “default” group was editable, but sure enough after clicking on it you can configure which repositories and whether it accepts public repositories on the “Default” runner group.</p>
<p>Thanks <a class="mention" href="/dcorbin-wintellect">@dcorbin-wintellect</a></p>
